nixos/modules/programs/envision.nix

```diff
@@ -0,0 +1,55 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.envision;
+in
+{
+
+  options = {
+    programs.envision = {
+      enable = lib.mkEnableOption "envision";
+
+      package = lib.mkPackageOption pkgs "envision" {};
+
+      openFirewall = lib.mkEnableOption "the default ports in the firewall for the WiVRn server" // {
+        default = true;
+      };
+
+      highPriority = lib.mkEnableOption "high priority capability" // {
+        default = true;
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.avahi = {
+      enable = true;
+      publish = {
+        enable = true;
+        userServices = true;
+      };
+    };
+
+    security.wrappers."envision" = lib.mkIf cfg.highPriority {
+      setuid = false;
+      owner = "root";
```

Comment:
Are you sure this should be `root` here? Isn't this a GUI application? Why root access?

Reply:
@wegank Thoughts on this?

Reply:
`setuid` is false. The file has to have an owner, so `root` seems to be a sensible default that's hard to edit maliciously.

Reply:
I guess it's up to the author of the PR to explain, I have no idea what permission envision needs...